PubHub/PubHub.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using PubHub.API.Controllers.Problems;
+using PubHub.API.Domain;
+using PubHub.API.Domain.Auth;
+using PubHub.API.Domain.Identity;
+using PubHub.Common.Models.Accounts;
+
+namespace PubHub.API.Controllers
+{
+    [AllowAnonymous]
+    [ApiController]
+    [Route("[controller]")]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError, Type = typeof(ProblemDetails))]
+    public class AuthController : Controller
+    {
+        private readonly PubHubContext _context;
+        private readonly UserManager<Account> _userManager;
+        private readonly AuthService _authService;
+
+        public AuthController(PubHubContext context, UserManager<Account> userManager, AuthService authService)
+        {
+            _context = context;
+            _userManager = userManager;
+            _authService = authService;
+        }
+
+        [HttpPost("token")]
+        public async Task<IResult> GetTokenAsync([FromHeader] string email, [FromHeader] string password)
+        {
+            if (!ModelState.IsValid)
+            {
+                return Results.Problem(
+                    statusCode: BadRequestSpecification.STATUS_CODE,
+                    title: BadRequestSpecification.TITLE,
+                    detail: "Login request parameters were malformed.",
+                    extensions: new Dictionary<string, object?>
+                    {
+                        { "State", ModelState }
+                    });
+            }
+
+            // Validate user email.
+            var account = await _userManager.FindByEmailAsync(email);
+            if (account == null)
+            {
+                return Results.Problem(
+                    statusCode: UnauthorizedSpecification.STATUS_CODE,
+                    title: UnauthorizedSpecification.TITLE,
+                    detail: "Email doesn't exist.",
+                    extensions: new Dictionary<string, object?>
+                    {
+                        { "Email", email }
+                    });
```

Comment:
I don't think we should inform that the email does not exist.
I'd rather have us say that either email or password didn't exist, as I think it's a security risk to expose such information.

Reply:
I'll change that to be a single check.